### PR TITLE
bpo-39353: Deprecate the binhex module

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue39353

### CODE REVIEW

**Comment by vstinner** [comment]
cc @tiran

**Comment by vstinner** [comment]
@serhiy-storchaka: Does you expect any issue just by marking the module as deprecated in 3.9?

**Comment by serhiy-storchaka** [comment]
What is a benefit of deprecating it?

This module is a high-level interface to several functions in the `binascii` module. This is all macOS specific stuff, so you need to ask macOS experts. If this encoding no longer used in macOS, its support can be removed, with deprecating first. If there are no plans to remove, there is no reason to deprecate.

**Comment by serhiy-storchaka** [comment]
If deprecate the binhex module, I think that it would be better to deprecate corresponding functions in the binascii module in the same issue and PR, so it will be easier to find the original discussion.

**Comment by vstinner** [comment]
> If deprecate the binhex module, I think that it would be better to deprecate corresponding functions in the binascii module in the same issue and PR, so it will be easier to find the original discussion.

Ok. Let's do that. I added a second commit which deprecates the 5 related functions of the binascii module.

I'm not sure about the binhex code to avoid emitting warnings on calls to deprecated binascii functions: `with _ignore_deprecation_warning():`. Is it useful to bother with that?

> What is a benefit of deprecating it?

Let me answer on the issue instead.

**Review by ronaldoussoren — APPROVED** [review]
LGTM

File: Doc/whatsnew/3.9.rst [hunk 1]
```diff
@@ -350,6 +350,16 @@ Deprecated
   deprecated and will be removed in version 3.11.
   (Contributed by Yury Selivanov and Kyle Stanley in :issue:`34790`.)
 
+* binhex4 and hexbin4 standards are now deprecated. The :`binhex` module
+  and the following :mod:`binascii` modules are now deprecated:
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.9.rst]:
```suggestion
  and the following :mod:`binascii` functions are now deprecated:
```

File: Lib/test/test_binascii.py [hunk 2]
```diff
@@ -416,6 +423,21 @@ def test_b2a_base64_newline(self):
         self.assertEqual(binascii.b2a_base64(b, newline=False),
                          b'aGVsbG8=')
 
+    def test_deprecated_warnings(self):
+        with support.check_warnings(('', DeprecationWarning)):
```
Comment (serhiy-storchaka) [inline Lib/test/test_binascii.py]: It is better to use standard `assertWarns()`.

File: Lib/test/test_binhex.py [hunk 3]
```diff
@@ -3,10 +3,12 @@
    Uses the mechanism of the python binhex module
    Based on an original test by Roger E. Masse.
 """
-import binhex
 import unittest
 from test import support
 
+with support.check_warnings(('', DeprecationWarning)):
+    import binhex
```
Comment (serhiy-storchaka) [inline Lib/test/test_binhex.py:10]: Is `binhex` imported in `test___all__` or other tests?

Comment (vstinner) [reply Lib/test/test_binhex.py:10]: The full test suite pass with -Werror, except of test_signal which is unrelated to this PR: https://bugs.python.org/issue39424

File: Lib/test/test_binascii.py (hunk 2 above)
Comment (vstinner) [reply Lib/test/test_binascii.py]: Done

File: Doc/whatsnew/3.9.rst (hunk 1 above)
Comment (vstinner) [reply Doc/whatsnew/3.9.rst]: Ooops, fixed!

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
@serhiy-storchaka: Thank you for insisting to also deprecate related binascii functions. You're right that it's more consistent.

**Comment by bugadani** [comment]
Can someone suggest an alternative to `crc_hqx` that provides the same result with the same or better performance?

**Pull request merged**